draft-ietf-httpbis-message-signatures.md

```diff
@@ -1048,6 +1051,49 @@ Signature: sig1=:P0wLUszWQjoi54udOtydf9IWTfNhy+r53jGFj9XZuP4uKwxyJo\
 
 The proxy's signature and the client's original signature can be verified independently for the same message, depending on the needs of the application.
 
+# Requesting Signatures
+
+While a signer is free to attach a signature to a request or response without prompting, it is often desirable for a potential verifier to signal that it expects a signature from a potential signer using the `Accept-Signature` field.
+
+The message to which the requested signature is applied is known as the "target message". When the `Accept-Signature` field is sent in an HTTP Request message, the field indicates that the client desires the server to sign the response using the identified parameters and the target message is the response to this request. All responses from resources that support such signature negotiation SHOULD either be uncacheable or contain a `Vary` header field that lists `Accept-Signature`, in order to prevent a cache from returning a response with a signature intended for a different request.
+
+When the `Accept-Signature` field is used in an HTTP Response message, the field indicates that the server desires the client to sign its next request to the server with the identified parameters, and the target message is the client's next request. The client can choose to also continue signing future requests to the same server in the same way.
```

Comment:
@mnot is it ok to include the language about a client choosing to sign future requests, or is that too stateful?

Reply:
This is fine; the state is associated with the resource, which *is* stateful. It's connection state we have to watch out for.

Reply:
Is it the next request specifically, or future requests more broadly?  This seems to presuppose the server knows the clients order of requests.